components/inputs/input-number.js

```diff
@@ -1,20 +1,52 @@
 import './input-text.js';
 import { css, html, LitElement } from 'lit-element/lit-element.js';
-import { formatNumber, parseNumber } from '@brightspace-ui/intl/lib/number.js';
+import { formatNumber, getNumberDescriptor, parseNumber } from '@brightspace-ui/intl/lib/number.js';
 import { FormElementMixin } from '../form/form-element-mixin.js';
 import { getUniqueId } from '../../helpers/uniqueId.js';
 import { ifDefined } from 'lit-html/directives/if-defined.js';
 import { LocalizeCoreElement } from '../../lang/localize-core-element.js';
 import { SkeletonMixin } from '../skeleton/skeleton-mixin.js';
 
-function formatValue(value, minFractionDigits, maxFractionDigits) {
+function formatValue(value, minFractionDigits, maxFractionDigits, showTrailingZeroes, numDecimalDigits) {
+	let adjustedMinFractionDigits = minFractionDigits;
+	if (showTrailingZeroes && numDecimalDigits > 0) {
```

Comment:
To make trailing zeroes work, we pass in an overridden `minFractionDigits` to our number formatting code based on how many digits they typed in beyond the decimal. This code here just makes sure that we're still not exceeding the `max-fraction-digits` setting (if set).